students/KennanY/Lesson6/MailRoomPt4.py

```diff
@@ -0,0 +1,137 @@
+
+        #Write to file
+    try:
+        with open(filename, 'w') as file:
+            letter='Dear {}, \n\n\tThank for for your very kind donations of ${}.\n\n\tIt will be put to very ' \
+                   'good use.\n\n\t\tSincerely, \n\n\t\t\t - The Team'.format(donor,
+                                                                                                       str(amount[len(amount)-1]))
+            file.write(letter)
+    except OSError as fileerr:
+        print(fileerr)
+        print("\nThere was an error opening the file.\n")
+        return
+    except FileNotFoundError as fnferr:
+        print(fnferr)
+        print("\nThere was an error reading the file")
+        return
+    except ValueError as verr:
+        print("There was an error writing to the file.")
+        return
+
+def func_Send_Thankyou():
+    ''' Send a Thank you note'''
+    print('Thank you')
+    value=0
+    name = input('Type donor name or ''list'' to see all the names=')
+    if name == 'list':
+        # List all donors if user enters 'list'
+        for donor in people.keys():
+            print(donor)
+    else:  # It is assumed that name entered in this case
+        # Prompt for donation amount
+        try:
+            value = float(input('Please enter the donation amount='))
+            # Is it in the db?
+            # If in database, append value else create a new record
+            if name in people:
+                #Add the donation
+                getlist = people[name]
+                getlist.append(value)
+        except:
+            print("Invalid value entered!")
+            return
+        else:
+            #Add the name and the donation
+            people[name] = value
+        # Print the email now
+        PrintEMail(name, value)
+
+def func_Create_Report():
+    """Print donation history"""
+    donations= []
+    for donors in people:
+        entry=[]
+        entry.append(donors)
+        for item in people[donors]:
+            entry.append(item)
+
+        donations.append(entry)
+
+    #Sort the donations
+    report_data = [[person[0], sum(person[1:]), len(person[1:]), sum(person[1:]) / len(person[1:])]
+                  for person in donations]
+    sorted_data = sorted(report_data, key=lambda x: x[1], reverse=True)
+
+    #Print the header
+    print('{0:25}{1:15}{2:15}{3:15}'.format('Donor Name', '| Total Gifts', '| Num Gifts', '| Average Gift'))
+    print('=' * 50)
+
+    for donor in donations:
+        total_amount=sum(donor[1:])
+        total_donation=len(donor)-1
+        avg_donation=total_amount/total_donation
+        print('{:<25s}  ${:>18,.2f}  {:>15d}  ${:>18,.2f}'.format( donor[0], total_amount, total_donation,
+                                                                      avg_donation))
+
+def  func_Send_Letters():
+    print('Write the letters to file in ' + os.getcwd())
+    Generate_Letters(people)
+
+def func_Quit():
+    global flag
+    flag=False
```

Comment:
This is interesting, but it's setting the flag to False so that the program will not run again once the user quits.  Is this what you intended?